File: Spring/DemoSpringBootApp/src/main/java/net/eikatou/springboot/HelloController.java
package net.eikatou.springboot;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.servlet.ModelAndView;

@Controller
public class HelloController {
	
	@RequestMapping(value="/", method=RequestMethod.GET)
	public ModelAndView index(ModelAndView mav) {
		mav.setViewName("index");
		mav.addObject("msg", "input your name:");
		return mav;
	}
	
	@RequestMapping(value="/", method=RequestMethod.POST)
	public ModelAndView send(@RequestParam("name")String name,
			ModelAndView mav) {
		mav.setViewName("index");
		mav.addObject("msg", "Hello " + name + " !");
		mav.addObject("value", name);
		return mav;
	}
}

File: Spring/SQSApp/src/main/kotlin/com/example/demo/controller/HelloController.kt
package com.example.demo.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.cloud.aws.messaging.core.QueueMessagingTemplate
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("api/hello")
class HelloController {
    @Autowired
    @Qualifier("friendQueueMessaging")
    lateinit var sqsAppQueueMessaging: QueueMessagingTemplate

    @RequestMapping(method = [RequestMethod.GET])
    fun hello(): String {
        return sqsAppQueueMessaging.receiveAndConvert(String::class.java) ?: ""
    }

    @RequestMapping(method = [RequestMethod.POST])
    @ResponseStatus(HttpStatus.CREATED)
    fun postMessage(@RequestBody message: String): String {
        sqsAppQueueMessaging.convertAndSend(message)
        return "OK"
    }

}